stackslib/src/net/api/getinfo.rs

```diff
@@ -217,16 +221,46 @@ impl RPCRequestHandler for RPCPeerInfoRequestHandler {
         node: &mut StacksNodeState,
     ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
         let ibd = node.ibd;
-        let rpc_peer_info =
-            node.with_node_state(|network, _sortdb, chainstate, _mempool, rpc_args| {
-                RPCPeerInfoData::from_network(
+
+        let rpc_peer_info: Result<RPCPeerInfoData, StacksHttpResponse> =
+            node.with_node_state(|network, sortdb, chainstate, _mempool, rpc_args| {
+                let header = self
+                    .get_stacks_chain_tip(&preamble, sortdb, chainstate)
+                    .map_err(|e| {
+                        StacksHttpResponse::new_error(
+                            &preamble,
+                            &HttpServerError::new(format!("Failed to load chain tip: {:?}", &e)),
+                        )
+                    })?;
+
+                let coinbase_height = NakamotoChainState::get_coinbase_height(
+                    &mut chainstate.index_conn(),
+                    &header.index_block_hash(),
```

Comment:
The network stack keeps a cached view of the latest stacks tip, which you can use to get this `index_block_hash` instead of fetching a header. 
```suggestion
                    &StacksBlockId::new(&network.stacks_tip.consensus_hash, &network.stacks_tip.block_hash),
```